tests/test_page_inputs.py

```diff
@@ -189,12 +192,20 @@ def test_http_request_init_full():
     http_body = HttpRequestBody(b"body")
     req_2 = HttpRequest("url", method="POST", headers=http_headers, body=http_body)
 
-    assert req_1.url == req_2.url
+    assert str(req_1.url) == str(req_2.url)
     assert req_1.method == req_2.method
     assert req_1.headers == req_2.headers
     assert req_1.body == req_2.body
 
 
+def test_http_request_init_with_response_url():
+    resp = HttpResponse("url", b"")
+    assert isinstance(resp.url, ResponseUrl)
+    req = HttpRequest(resp.url)
+    assert isinstance(req.url, RequestUrl)
+    assert str(req.url) == str(resp.url)
+
```

Comment:
@Gallaecio has raised a good point if `HttpResponse("url")` and `HttpRequest("url")` are equal. We should add a test for such.

Reply:
I've added a test which shows that they're not equal.

But I'm not sure this behavior is useful. What kind of errors would it prevent?

I wonder if it'd better to leave it "undefined", instead of fixing it in tests (if that's even a reasonable approach :)

pathlib does some magic when comparing paths, it's not a simple string match. E.g. on Windows path components are lower-cased before comparison.

Reply:
The test simply solidifies the expectation to users in case they're unsure. As a user, I might expect when initially using it for the first time that the expression `response.url == request.url` holds True. However, I need to cast it to its string value to bear True.

I also expect such test to conveniently break when we override `__eq__` to another behavior. It helps future PRs denote that an equality expectation will change.